platform/core/src/services/ToolBarService/ToolbarService.ts

```diff
@@ -130,160 +136,182 @@ export default class ToolbarService extends PubSubService {
-
-          // only set the primary tool if no error was thrown.
-          // if the itemId is not undefined use it; otherwise, set the first tool in
-          // the commands as the primary tool
-          this.state.primaryToolId = itemId || commands[0].commandOptions?.toolName;
-        } catch (error) {
-          console.warn(error);
-        }
-
-        break;
-      }
-      case 'toggle': {
-        const { commands } = interaction;
-        let commandExecuted;
-
-        // only toggle if a command was executed
-        this.state.toggles[itemId] =
-          this.state.toggles[itemId] === undefined ? true : !this.state.toggles[itemId];
-
-        if (!commands) {
-          break;
-        }
+    const itemId = interaction.itemId ?? interaction.id;
+    interaction.itemId = itemId;
 
-        commands.forEach(({ commandName, commandOptions, context }) => {
-          if (!commandOptions) {
-            commandOptions = {};
-          }
-
-          if (commandName) {
-            commandOptions.toggledState = this.state.toggles[itemId];
-
-            try {
-              commandsManager.runCommand(commandName, commandOptions, context);
-              commandExecuted = true;
-            } catch (error) {
-              console.warn(error);
-            }
-          }
-        });
-
-        if (!commandExecuted) {
-          // If no command was executed, we need to toggle the state back
-          this.state.toggles[itemId] = !this.state.toggles[itemId];
-        }
+    const commands = Array.isArray(interaction.commands)
+      ? interaction.commands
+      : [interaction.commands];
 
-        break;
-      }
-      default:
-        throw new Error(`Invalid interaction type: ${interactionType}`);
+    if (!commands?.length) {
+      this.refreshToolbarState({
+        ...options?.refreshProps,
+        itemId,
+        interaction,
+      });
     }
 
-    // Todo: comment out for now
-    // Run command if there's one associated
-    //
-    // NOTE: Should probably just do this for tools as well?
-    // But would be nice if we could enforce at least the command name?
-    // let unsubscribe;
-    // if (commandName) {
-    //   unsubscribe = commandsManager.runCommand(commandName, commandOptions);
-    // }
-
-    // // Storing the unsubscribe for later resetting
-    // if (unsubscribe && typeof unsubscribe === 'function') {
-    //   if (this.unsubscriptions.indexOf(unsubscribe) === -1) {
-    //     this.unsubscriptions.push(unsubscribe);
-    //   }
-    // }
-
-    // Track last touched id for each group
-    if (groupId) {
-      this.state.groups[groupId] = itemId;
-    }
+    const commandOptions = { ...options, ...interaction };
 
-    this._broadcastEvent(this.EVENTS.TOOL_BAR_STATE_MODIFIED, { ...this.state });
-  }
+    // Loop through commands and run them with the combined options
+    commands.forEach(command => {
+      this._commandsManager.run(command, commandOptions);
```

Comment:
this._commandsManager.run(commands, commandOptions) is sufficient.

Reply:
Done